server/src/main/java/org/opensearch/cluster/ClusterModule.java

```diff
@@ -214,6 +215,8 @@ public static List<Entry> getNamedWriteables() {
             DecommissionAttributeMetadata::new,
             DecommissionAttributeMetadata::readDiffFrom
         );
+
+        registerMetadataCustom(entries, QueryGroupMetadata.TYPE, QueryGroupMetadata::new, QueryGroupMetadata::readDiffFrom);
```

Comment:
Shall we add test for validating this? It was missing in the previous revision and all tests were passing

Reply:
I did not see any tests for NamedWritables. But I think it shouldn;t be hard to assert the presence of the new entity